feat: add `math/base/special/logitf`

vivekmaurya001 opened the pull request:
Progresses #649 

## Description

- adds math/base/special/logitf, which would be the single-precision equivalent for [math/base/special/logit](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/logit).

This pull request:

- Progresses: https://github.com/stdlib-js/stdlib/issues/649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)


* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

vivekmaurya001 committed:
feat(math): add math/base/special/logitf

vivekmaurya001 committed:
Merge branch 'logitf' of https://github.com/vivekmaurya001/stdlib into logitf

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/src/main.c:
```diff
@@ -0,0 +1,50 @@
+*/
+
+#include "stdlib/math/base/special/logitf.h"
+#include "stdlib/math/base/assert/is_probability.h"
```
@vivekmaurya001 We'll need to use `math/base/assert/is_probabilityf` here, which needs to be implemented first.

vivekmaurya001 replied:
ook i will implement that first , thanks for pointing out

vivekmaurya001 replied:
i have changed it !

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into logitf

vivekmaurya001 committed:
add math/base/special/logitf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

vivekmaurya001 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

vivekmaurya001 committed:
add math/base/special/logitf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

vivekmaurya001 committed:
Merge branch 'logitf' of https://github.com/vivekmaurya001/stdlib into logitf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

vivekmaurya001 commented:
Hi @gunjjoshi  , i have made the changes according to recent PR discussions , can u please check

vivekmaurya001 committed:
add math/base/special/logitf

vivekmaurya001 committed:
add math/base/special/logitf

vivekmaurya001 commented:
@gunjjoshi , @Planeshifter , Is this PR need some changes or is it good to go ?

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/README.md:
```diff
@@ -0,0 +1,216 @@
+
+# logitf
+
+> Compute the [logit][logit] function for single precision-floating point number.
```
```suggestion
> Compute the [logit][logit] function for a single-precision floating-point number.
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/README.md:
```diff
@@ -0,0 +1,216 @@
+
+#### logitf( p )
+
+Computes the [logit][logit] function for single precision-floating point number.
```
```suggestion
Computes the [logit][logit] function for a single-precision floating-point number.
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/README.md:
```diff
@@ -0,0 +1,216 @@
+var opts = {
+    'dtype': 'float32'
+};
+var p = uniform( len, 0, 1, opts );
```
```suggestion
var p = uniform( len, 0.0, 1.0, opts );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/README.md:
```diff
@@ -0,0 +1,216 @@
+var i;
+
+for ( i = 0; i < 100; i++ ) {
+    console.log( 'logitf(%f) = %f', p[ i ], logitf( p[ i ] ) );
```
```suggestion
    console.log( 'logitf(%d) = %d', p[ i ], logitf( p[ i ] ) );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/README.md:
```diff
@@ -0,0 +1,216 @@
+
+#### stdlib_base_logitf( p )
+
+Computes the [logit][logit] function for single precision-floating point number.
```
```suggestion
Computes the [logit][logit] function for a single-precision floating-point number.
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/README.md:
````diff
@@ -0,0 +1,216 @@
+Computes the [logit][logit] function for single precision-floating point number.
+
+```c
+float out = stdlib_base_logitf( 0.2 );
````
```suggestion
float out = stdlib_base_logitf( 0.2f );
```

gunjjoshi replied:
Missing `f` suffixes here and below.

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/README.md:
````diff
@@ -0,0 +1,216 @@
+
+```c
+float out = stdlib_base_logitf( 0.2 );
+// returns ~-1.386
````
```suggestion
// returns ~-1.386f
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/README.md:
```diff
@@ -0,0 +1,216 @@
+float out = stdlib_base_logitf( 0.2 );
+// returns ~-1.386
+
+out = stdlib_base_logitf( 0.9 );
```
```suggestion
out = stdlib_base_logitf( 0.9f );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/README.md:
```diff
@@ -0,0 +1,216 @@
+// returns ~-1.386
+
+out = stdlib_base_logitf( 0.9 );
+// returns ~2.197
```
```suggestion
// returns ~2.197f
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/README.md:
```diff
@@ -0,0 +1,216 @@
+    int i;
+
+    for ( i = 0; i < 100; i++ ) {
+        x = ( float )rand() / ( float )RAND_MAX;
```
```suggestion
        x = (float)rand() / (float)RAND_MAX;
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/benchmark/benchmark.js:
```diff
@@ -0,0 +1,59 @@
+	};
+
+	len = 100;
+	x = uniform( len, 0, 1, opts );
```
```suggestion
	x = uniform( len, 0.0, 1.0, opts );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,68 @@
+	};
+
+	len = 100;
+	x = uniform( len, 0, 1, opts );
```
```suggestion
	x = uniform( len, 0.0, 1.0, opts );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,152 @@
+#include <time.h>
+#include <sys/time.h>
+
+#define NAME "logit"
```
```suggestion
#define NAME "logitf"
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,152 @@
+*
+* @return random number
+*/
+static double rand_double( void ) {
```
```suggestion
static float rand_float( void ) {
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,152 @@
+*/
+static double rand_double( void ) {
+	int r = rand();
+	return (double)r / ( (double)RAND_MAX + 1.0 );
```
```suggestion
	return (float)r / ( (float)RAND_MAX + 1.0f );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,152 @@
+}
+
+/**
+* Evaluates the logit function.
```
```suggestion
* Computes the logit function for a single-precision floating-point number.
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,152 @@
+* @param x   input value
+* @return result
+*/
+double logit( double x ) {
+	if ( x != x ) {
+		return x;
+	}
+	if ( x < 0.0 || x > 1.0 ) {
+		return NAN;
+	}
+	if ( x == 0.0 ) {
+		return -HUGE_VAL;
+	}
+	if ( x == 1.0 ) {
+		return HUGE_VAL;
+	}
+	return log( x / ( 1.0 - x ) );
+}
```
@vivekmaurya001 This must be modified for single-precision.

vivekmaurya001 replied:
Oh, sorry I didn't notice it

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,152 @@
+*/
+static double benchmark( void ) {
+	double elapsed;
+	double y;
+	double t;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		y = logit( rand_double() );
+		if ( y != y ) {
+			printf( "should not return NaN\n" );
+			break;
+		}
+	}
+	elapsed = tic() - t;
+	if ( y != y ) {
+		printf( "should not return NaN\n" );
+	}
+	return elapsed;
```
Same goes for this too.

vivekmaurya001 replied:
ook

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/benchmark/c/native/benchmark.c:
```diff
@@ -0,0 +1,136 @@
+	double elapsed;
+	float x[ 100 ];
+	float y;
+	double t;
```
```suggestion
	double t;
	float y;
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/docs/repl.txt:
```diff
@@ -0,0 +1,35 @@
+
+{{alias}}( p )
+    Evaluates the logit function for single precision-floating point number.
```
```suggestion
    Evaluates the logit function for a single-precision floating-point number.
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/docs/types/index.d.ts:
```diff
@@ -0,0 +1,57 @@
+// TypeScript Version: 4.1
+
+/**
+* Evaluates the logit function for single precision-floating point number.
```
```suggestion
* Evaluates the logit function for a single-precision floating-point number.
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/examples/index.js:
```diff
@@ -0,0 +1,33 @@
+var opts = {
+	'dtype': 'float32'
+};
+var p = uniform( len, 0, 1, opts );
```
```suggestion
var p = uniform( len, 0.0, 1.0, opts );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/examples/index.js:
```diff
@@ -0,0 +1,33 @@
+var i;
+
+for ( i = 0; i < 100; i++ ) {
+	console.log( 'logitf(%f) = %f', p[ i ], logitf( p[ i ] ) );
```
```suggestion
	console.log( 'logitf(%d) = %d', p[ i ], logitf( p[ i ] ) );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/include/stdlib/math/base/special/logitf.h:
```diff
@@ -0,0 +1,38 @@
+#endif
+
+/**
+* Evaluates the logit function for single precision-floating point number.
```
```suggestion
* Evaluates the logit function for a single-precision floating-point number.
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/lib/index.js:
```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+/**
+* Evaluate the logit function for single precision-floating point number.
```
```suggestion
* Evaluate the logit function for single-precision floating-point number.
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/lib/main.js:
```diff
@@ -0,0 +1,77 @@
+// MAIN //
+
+/**
+* Evaluates the logit function for single precision-floating point number.
```
```suggestion
* Evaluates the logit function for a single-precision floating-point number.
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/lib/native.js:
```diff
@@ -0,0 +1,62 @@
+// MAIN //
+
+/**
+* Evaluates the logit function for single precision-floating point number.
```
```suggestion
* Evaluates the logit function for a single-precision floating-point number.
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/package.json:
```diff
@@ -0,0 +1,66 @@
+{
+  "name": "@stdlib/math/base/special/logitf",
+  "version": "0.0.0",
+  "description": "Logit function for single precision-floating point number.",
```
@vivekmaurya001 This doesn't match with the description used elsewhere.

vivekmaurya001 replied:
updating it too

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/src/main.c:
```diff
@@ -0,0 +1,50 @@
+#include "stdlib/constants/float32/ninf.h"
+
+/**
+* Evaluates the logit function for single precision-floating point number.
```
```suggestion
* Evaluates the logit function for a single-precision floating-point number.
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/src/main.c:
```diff
@@ -0,0 +1,50 @@
+* Evaluates the logit function for single precision-floating point number.
+*
+* @param x    input value
+* @return	  output value
```
```suggestion
* @return     output value
```

gunjjoshi replied:
Mix of tabs and spaces.

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/src/main.c:
```diff
@@ -0,0 +1,50 @@
+* @return	  output value
+*
+* @example
+* float out = stdlib_base_logitf( 0.2 );
+* // returns ~-1.386
```
```suggestion
* float out = stdlib_base_logitf( 0.2f );
* // returns ~-1.386f
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/test/fixtures/python/runner.py:
```diff
@@ -0,0 +1,80 @@
+    x = np.linspace(0.0001, 0.25, 500).astype(np.float32)
+    gen(x, "small.json")
+
+    x = np.linspace(0.25, 0.49, 500).astype(np.float32)
+    gen(x, "medium_below.json")
+
+    x = np.linspace(0.5, 0.75, 500).astype(np.float32)
+    gen(x, "medium_above.json")
```
@vivekmaurya001 Any specific reason to break down the range `0.25 - 0.75` into 2 parts? In `logit`, we are having a single range for this.

vivekmaurya001 replied:
Yeah, actually what happened is medium test case almost all tests   got passed except 3 with tolerance below 20 and for these 3 tests tolerance is taking about 250 value, so I thought to split it

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/logitf/test/test.js:
```diff
@@ -0,0 +1,164 @@
+			t.equal( y, expected[ i ], 'x: '+x[ i ]+', y: '+y+', expected: '+expected[ i ] );
+		} else {
+			delta = absf( y - expected[ i ] );
+			tol = EPS * 15 * absf( expected[ i ] );
```
```suggestion
			tol = 15.0 * EPS * absf( expected[ i ] );
```

gunjjoshi replied:
This change can be applied here and everywhere else in this PR @vivekmaurya001 .

vivekmaurya001 replied:
done here too

gunjjoshi committed:
Apply suggestions from code review

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi committed:
Update lib/node_modules/@stdlib/math/base/special/logitf/docs/types/index.d.ts

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

gunjjoshi commented:
Tested this PR locally. The tolerance of `15.0 * EPS` isn't required in `test.native.js`, for the C implementation. But, that is required in `test.js`, for the JS implementation. Any reason to keep the same tolerance level of `15.0 * EPS` in `test.native.js` as well, even though it wasn't required, @vivekmaurya001?

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into logitf

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into logitf

vivekmaurya001 commented:
@gunjjoshi, I didnt realised it I thought the test file was requiring so this file also required it at that time while testing native file got skipped, sorry about that

vivekmaurya001 committed:
feat: add math/base/special/logitf

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: passed
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

vivekmaurya001 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

anandkaranubc commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into logitf

anandkaranubc committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: fix function descriptions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
bench: follow code conventions and fix function return types

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: replace manual for loop in examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: re-enable lint rule

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
fix: use float64ToFloat32 to avoid elevated precision

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
test: replace equal with strictEqual

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/logitf/lib/main.js:
```diff
@@ -0,0 +1,78 @@
+	if ( p === 1.0 ) {
+		return PINF;
+	}
+	return lnf( float64ToFloat32( p / float64ToFloat32( 1.0-p ) ) );
```
The need for a higher tolerance threshold in the JS tests is no longer necessary with this change. Previously, without using `float64ToFloat32`, calculations were elevated to double precision, resulting in different outputs.

anandkaranubc reviewed (approved):
Good to go! Thanks @vivekmaurya001 and @gunjjoshi!

Planeshifter reviewed (approved):
Let's get it in. Thanks, everyone!
